sklearn/linear_model/ridge.py

```diff
@@ -1046,10 +1046,12 @@ def fit(self, X, y, sample_weight=None):
         error = scorer is None
 
         for i, alpha in enumerate(self.alphas):
+            if alpha < 0:
```

Comment:
The checking needs to be done outside from the loop. Otherwise we start to compute some stuff to actually break it at the end.

So something like:

```python
if np.any(alphas < 0):
    raise ValueError("alphas cannot be negative. Got {} containing some negative value instead.")
```